### ISSUE TITLE
add more of the javascript array functions

### ISSUE THREAD

**Issue by RedMan13** [scratch-for-discord/scratch-for-discord#180]
i am specifically focusing on
`filter()`
`find()` and `findLast()`
`findIndex()` and `findLastIndex()`
`forEach()` a replacment for the current for loop
`map()`
`reduce()` and `reduceRight()`
`sort()`
`concat()`

you can find the rest of the array functions [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array)

**Comment by noxyyk** [comment]
at()
unshift()
shift()
pop()

**Comment by RedMan13** [comment]
those other then at are already suplied by default list blocks ._.

**Comment by RedMan13** [comment]
or atliest i remember them being so

**Comment by JeremyGamer13** [comment]
> at() unshift() shift() pop()

all (except maybe at()) of these are already in the Lists category

**Comment by RedMan13** [comment]
yeah, at is usful tho for accessing from the end of a list rather then from the start
